client/layout/index.jsx

```diff
@@ -184,11 +184,7 @@ class Layout extends Component {
 				{ this.props.isOffline && <OfflineStatus /> }
 				<div id="content" className="layout__content">
 					{ config.isEnabled( 'jitms' ) && this.props.isEligibleForJITM && (
-						<AsyncLoad
-							require="calypso/blocks/jitm"
-							placeholder={ null }
-							messagePath={ `calypso:${ this.props.sectionJitmPath }:admin_notices` }
-						/>
+						<AsyncLoad require="calypso/blocks/jitm" messagePathSuffix="admin_notices" />
```

Comment:
This change introduces a bug. For routes like `/themes/upload` or `/plugins/upload`, the `messagePath` used to be `calypso:themes-upload:admin_notices` and `calypso:plugins-upload:admin_notices`, respectively. The `themes-upload` slug was coming from the `getMessagePathForJITM` selector.

After this change, the `messagePath` is based on section name, the slug becoming `themes` and `plugins`.

The `themes-upload` and `plugins-upload` slugs are really used by the server and specific messages are generated:

<img width="1134" alt="Screenshot 2020-10-27 at 08 21 01" src="https://user-images.githubusercontent.com/664258/97269183-6579ad00-182d-11eb-970d-b8dc5aea98b3.png">

This place should use the `JITMConnected` variant exported from the `block/jitm` module, and supply a custom `messagePath`.

The trouble is that `AsyncLoad` can use only default exports of a module 🙁 An elegant solution would be to change the API of the `blocks/jitm` component: if the `messagePath` prop is supplied, use it, and generate one from `messagePathSuffix` only if it's not present:
```js
connect( ( state, { messagePath, messagePathSuffix } ) => {
  if ( messagePath === undefined ) {
    messagePath = `calypso:${ getSectionName( state ) }:${ messagePathSuffix }`;
  }
  return { messagePath };
} )
```
This is a fairly common Redux pattern.

Reply:
Created #46821

Reply:
PR: #46824